Unity 5 Support

quasmon opened the pull request:
Hey there,

I ran into a few issues while getting our projects working with Unity 5 and wanted to share the fixes as some people might find them helpful:
- When an Xcode project was loaded by Unity (during an append) we would see an exception being thrown because the path values that contained "-" for some file references were not quote escaped. 
- Unity seems to add file references with no build phase which meant when adding a projmod addition for e.g. AdSupport.framework it was ignored. Had to make some changes to allow phase changes and also detecting duplicate build files being added.

Feel free to let me know if you need anything changed.

Cheers,
Ramon.

quasmon committed:
Quote escape any string values that contain a "-" character

Unity 5 expects string values that contain "-" characters to be quote escaped like Xcode does

quasmon committed:
Handle file references which now contain a valid build phase

Previously file references that already existed but had a build phase set would be ignored. Unity 5 generates file references e.g AdSupport.framework that don't belong to any build phase. Change the way file references are added so that a change to a build phase will add that file to the required build phase. Required changing the build references to not allow duplicates

quasmon committed:
Make all projmod modifications optional

quasmon committed:
Allow changes to weak property when it already exists

Was throwing an exception because it already exists

gforeman commented:
in the regex in PBXParse.cs, shouldn't you remove the escape character "/"
when removing the "-"?

On Thu, Mar 26, 2015 at 9:40 AM, Ramon Steenson <notifications@github.com>
wrote:

> Hey there,
>
> I ran into a few issues while getting our projects working with Unity 5
> and wanted to share the fixes as some people might find them helpful:
>
>    - When an Xcode project was loaded by Unity (during an append) we
>    would see an exception being thrown because the path values that contained
>    "-" for some file references were not quote escaped.
>    - Unity seems to add file references with no build phase which meant
>    when adding a projmod addition for e.g. AdSupport.framework it was ignored.
>    Had to make some changes to allow phase changes and also detecting
>    duplicate build files being added.
>
> Feel free to let me know if you need anything changed.
>
> Cheers,
> Ramon.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/onevcat/XUPorter/pull/48
> Commit Summary
>
>    - Quote escape any string values that contain a "-" character
>    - Handle file references which now contain a valid build phase
>    - Make all projmod modifications optional
>    - Allow changes to weak property when it already exists
>
> File Changes
>
>    - *M* PBX Editor/PBXBuildFile.cs
>    <https://github.com/onevcat/XUPorter/pull/48/files#diff-0> (9)
>    - *M* PBX Editor/PBXParser.cs
>    <https://github.com/onevcat/XUPorter/pull/48/files#diff-1> (5)
>    - *M* XCProject.cs
>    <https://github.com/onevcat/XUPorter/pull/48/files#diff-2> (185)
>
> Patch Links:
>
>    - https://github.com/onevcat/XUPorter/pull/48.patch
>    - https://github.com/onevcat/XUPorter/pull/48.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/onevcat/XUPorter/pull/48>.
>



-- 
Gary J. Foreman
Founder | Slipstream Games

quasmon committed:
Fix compiler flags specified in the files projmod section not being
applied correctly. Also move compiler flags from file references to
build files since thats where they are saved

quasmon committed:
Change build target to iOS for Unity 5

Still support Unity 4.x though

quasmon committed:
Remove some logging

Don’t need it printing out for other platforms

onevcat commented:
Hi, @quasmon 

Sorry for the late reply.

Thanks for the p-r, it looks great to me.

Unity 5 has its own dependency system to add frameworks and third party files easier than Unity 4. So the priority to support Unity 5 is not so high currently. But I will take a look at it and try to resolve the conflict and merge it later.

Brantone commented:
I like that https://github.com/onevcat/XUPorter/pull/48/commits/555b2a4ee87c23f4feb05e98575d3222a901d060 solves https://github.com/onevcat/XUPorter/issues/38 ... how hard would it be to update that for folders as well?

Also, although Unity 5 has incorporated some libraries, it's still not as ease to use like projmods for things like lproj folders.
